packages/react/src/solana/providers/ThirdwebSDKProvider.tsx

```diff
@@ -0,0 +1,72 @@
+import {
+  ConnectionProvider,
+  WalletProvider,
+  useWallet,
+  useConnection,
+} from "@solana/wallet-adapter-react";
+import { ThirdwebSDK } from "@thirdweb-dev/solana";
+import {
+  createContext,
+  FC,
+  PropsWithChildren,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+
+interface TWSolanaProviderProps {}
+
+/**
+ * Gives access to the ThirdwebSDK instance and other useful hooks to the rest of the app.
+ * Requires to be wrapped with a ConnectionProvider and a WalletProvider from @solana/wallet-adapter-react.
+ * @example
+ * ```tsx
+ * import {
+ *   ConnectionProvider,
+ *   WalletProvider,
+ * } from "@solana/wallet-adapter-react";
+ * import { ThirdwebProvider } from "@thirdweb-dev/react/solana";
+ *
+ * const App = () => {
+ *  return (
+ *      <ConnectionProvider endpoint={endpoint}>
+ *          <WalletProvider wallets={wallets} autoConnect>
+ *              <ThirdwebSDKProvider>
+ *                  <YourApp />
+ *              </ThirdwebProvider>
+ *          </WalletProvider>
+ *      </ConnectionProvider>
+ * )};
+ * ```
+ */
+export const ThirdwebSDKProvider: FC<
+  PropsWithChildren<TWSolanaProviderProps>
+> = ({ children }) => {
+  const { connection } = useConnection();
+  const wallet = useWallet();
```

Comment:
if we're doing this we should just do the full wrapping provider

Reply:
it's an extra 4-5 LOC at this point and now we're already pulling in the deps etc

Reply:
i had it as peer dep. but yeah

it would definitely make it simpler on the DX side, especially ppl who are not familiar with solana js

let me try it out